src/main/java/com/monumental/services/AsyncService.java

```diff
@@ -0,0 +1,28 @@
+package com.monumental.services;
+
+import com.monumental.util.async.AsyncJob;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This abstract class should be extended by asynchronous services that need to maintain
+ * an internal job queue in order to report progress and results of AsyncJobs as they run
+ */
+public abstract class AsyncService {
```

Comment:
This abstract class should be extended by async services and basically just maintains an internal map of jobs using their id, so that they can be retrieved later to check the progress or get the final result